src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -1082,14 +1088,24 @@ bool QgsPostgresProvider::loadFields()
   for ( int i = 0; i < result.PQnfields(); i++ )
   {
     QString fieldName = result.PQfname( i );
+    Oid tableoid = result.PQftable( i );
+    int attnum = result.PQftablecol( i );
+    QString formattedFieldType = fmtFieldTypeMap[tableoid][attnum];
     if ( fieldName == mGeometryColumn )
+    {
+      if ( mDetectedGeomType == Qgis::WkbType::Unknown )
+      {
+        // rare case, elevation can not be computed via mDetectedGeomType, we parse the formattedFieldType and look for Z data
+        const thread_local QRegularExpression re( QRegularExpression::anchoredPattern( "geometry\\(\\w+ZM?(,\\d+)?\\)" ), QRegularExpression::CaseInsensitiveOption );
```

Comment:
```suggestion
        const thread_local QRegularExpression re( QRegularExpression::anchoredPattern( "geometry\\(\\w+\s*Z?M?(,\\d+)?\\)" ), QRegularExpression::CaseInsensitiveOption );
```

Suggested changes:
- \s* : Sometimes WKT-like pattern is "POINT ZM" and not "POINTZM"
- Z?: Add a "?" here. Because, without "?", you only look for "Z" type. But, maybe it's normal?

Reply:
Applied. Thanks.

Reply:
> Z?: Add a "?" here. Because, without "?", you only look for "Z" type. But, maybe it's normal?

Yes it was intentional!